src/name.rs

```diff
@@ -3,13 +3,58 @@
 //!
 //! [spec]: https://www.w3.org/TR/xml-names11
 
-use crate::errors::{Error, Result};
 use crate::events::attributes::Attribute;
 use crate::events::BytesStart;
 use crate::utils::write_byte_string;
 use memchr::memchr;
 use std::fmt::{self, Debug, Formatter};
 
+/// Some namespace was invalid
+#[derive(Debug, Clone)]
+pub enum NamespaceError {
+    /// Specified namespace prefix is unknown, cannot resolve namespace for it
+    UnknownPrefix(Vec<u8>),
+    /// Error for when a reserved namespace is set incorrectly.
+    ///
+    /// This error returned in following cases:
+    /// - the XML document attempts to bind `xml` prefix to something other than
+    ///   `http://www.w3.org/XML/1998/namespace`
+    /// - the XML document attempts to bind `xmlns` prefix
+    /// - the XML document attempts to bind some prefix (except `xml`) to
+    ///   `http://www.w3.org/XML/1998/namespace`
+    /// - the XML document attempts to bind some prefix to
+    ///   `http://www.w3.org/2000/xmlns/`
+    InvalidPrefixBind {
+        /// The prefix that is tried to be bound
+        prefix: Vec<u8>,
+        /// Namespace to which prefix tried to be bound
+        namespace: Vec<u8>,
+    },
+}
+
+impl fmt::Display for NamespaceError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::UnknownPrefix(prefix) => {
+                f.write_str("Unknown namespace prefix '")?;
```

Comment:
As already mentioned, if we bring order in errors, make all texts with a small letter:
```suggestion
                f.write_str("unknown namespace prefix '")?;
```